app-office/visidata/visidata-2.1.1.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..9} )
+
+inherit distutils-r1 eutils
+
+DESCRIPTION="Terminal spreadsheet multitool for discovering and arranging data"
+HOMEPAGE="http://visidata.org"
+SRC_URI="https://github.com/saulpw/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```

Comment:
There are also some sphinx docs in `docs/api`, you can add support for these with the `distutils_enable_sphinx <deps>` function.

Reply:
I've tried adding this, but get the following errors - any ideas what I might be doing wrong?

```sh
orcing test.
Unable to configure loopback interface: Operation not permitted
 * visidata-2.1.1.tar.gz BLAKE2B SHA512 size ;-) ...                     [ ok ]
Unable to configure loopback interface: Operation not permitted
>>> Unpacking source...
>>> Unpacking visidata-2.1.1.tar.gz to /var/tmp/portage/app-office/visidata-2.1.1/work
>>> Source unpacked in /var/tmp/portage/app-office/visidata-2.1.1/work
Unable to configure loopback interface: Operation not permitted
>>> Preparing source in /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1 ...
>>> Source prepared.
Unable to configure loopback interface: Operation not permitted
>>> Configuring source in /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1 ...
>>> Source configured.
Unable to configure loopback interface: Operation not permitted
>>> Compiling source in /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1 ...
 * python3_8: running distutils-r1_run_phase distutils-r1_python_compile
python3.8 setup.py build -j 5
running build
running build_py
file visidata.py (for module visidata) not found
creating /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/__init__.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/__main__.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/_input.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/_open.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/_types.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/aggregators.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/basesheet.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/canvas.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/choose.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/clipboard.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/cliptext.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/cmdlog.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/color.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/colorsheet.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/column.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/customdate.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/deprecated.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/describe.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/editor.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/errors.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/expr.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/extensible.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/fill.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/freeze.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/freqtbl.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/graph.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/help.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/incr.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/join.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/layout.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/macros.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/main.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/mainloop.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/melt.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/menu.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/metasheets.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/misc.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/modify.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/motd.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/movement.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/path.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/pivot.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/plugins.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/pyobj.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/regex.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/save.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/search.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/selection.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/settings.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/sheets.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/shell.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/slide.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/sort.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/statusbar.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/textsheet.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/threads.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/transpose.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/undo.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/unfurl.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/urlcache.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/utils.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/vdobj.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
copying visidata/wrappers.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata
creating /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/__init__.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/_pandas.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/archive.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/csv.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/eml.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/fixed_width.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/frictionless.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/graphviz.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/hdf5.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/html.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/http.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/imap.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/json.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/markdown.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/mbtiles.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/mysql.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/npy.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/pandas_freqtbl.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/pcap.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/pdf.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/png.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/postgres.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/rec.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/sas.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/shp.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/spss.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/sqlite.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/texttables.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/tsv.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/ttf.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/usv.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/vcf.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/xlsb.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/xlsx.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/xml.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/xword.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
copying visidata/loaders/yaml.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/loaders
creating /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/tests
copying visidata/tests/__init__.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/tests
copying visidata/tests/conftest.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/tests
copying visidata/tests/test_commands.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/tests
copying visidata/tests/test_edittext.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/tests
copying visidata/tests/test_path.py -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/tests
running egg_info
creating visidata.egg-info
writing visidata.egg-info/PKG-INFO
writing dependency_links to visidata.egg-info/dependency_links.txt
writing entry points to visidata.egg-info/entry_points.txt
writing requirements to visidata.egg-info/requires.txt
writing top-level names to visidata.egg-info/top_level.txt
writing manifest file 'visidata.egg-info/SOURCES.txt'
file visidata.py (for module visidata) not found
reading manifest file 'visidata.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'visidata.egg-info/SOURCES.txt'
creating /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/man
copying visidata/man/vd.1 -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/man
copying visidata/man/vd.txt -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/lib/visidata/man
file visidata.py (for module visidata) not found
warning: build_py: byte-compiling is disabled, skipping.

running build_scripts
creating /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/scripts
copying and adjusting bin/vd -> /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/scripts
changing mode of /var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1-python3_8/scripts/vd from 644 to 755
 * python_setup: none of the enabled implementation matched the patterns.
 *   patterns: '(*)'
 * Likely a REQUIRED_USE constraint (possibly USE-conditional) is missing.
 *   suggested: || ( $(python_gen_useflags ) )
 * (remember to quote all the patterns with '')
 * ERROR: app-office/visidata-2.1.1::science failed (compile phase):
 *   python_setup: no enabled implementation satisfy requirements
 * 
 * Call stack:
 *     ebuild.sh, line  125:  Called src_compile
 *   environment, line 3064:  Called distutils-r1_src_compile
 *   environment, line 1172:  Called _distutils-r1_run_common_phase 'python_compile_all'
 *   environment, line  441:  Called python_setup
 *   environment, line 3012:  Called die
 * The specific snippet of code:
 *           die "${FUNCNAME}: no enabled implementation satisfy requirements";
 * 
 * If you need support, post the output of `emerge --info '=app-office/visidata-2.1.1::science'`,
 * the complete build log and the output of `emerge -pqv '=app-office/visidata-2.1.1::science'`.
 * The complete build log is located at '/var/tmp/portage/app-office/visidata-2.1.1/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/app-office/visidata-2.1.1/temp/environment'.
 * Working directory: '/var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1'
 * S: '/var/tmp/portage/app-office/visidata-2.1.1/work/visidata-2.1.1'
```

Reply:
It should be:
```
distutils_enable_sphinx docs/api \
	dev-python/recommonmark \
	dev-python/sphinx-argparse
```
instead of 
```
distutils_enable_sphinx docs \
	dev-python/recommonmark \
	dev-python/sphinx-argparse
```
Because the first argument should be the location of the `conf.py` file

Reply:
It looks like we're missing a dependency for the doc building:
```
Running Sphinx v2.4.4

Extension error:
Could not import extension sphinx_markdown_tables (exception: No module named 'sphinx_markdown_tables')
```
If you want you can package this (https://pypi.org/project/sphinx-markdown-tables/), it should be easy since that package only has 1 dependency, no test suite and no doc building of itself. If not, than what I usually do is just comment out the `distutils_enable_sphinx` lines, in any case don't remove them completely otherwise the information and the first steps to getting the doc building working would be completely gone.

Reply:
Done, I commented-out the `distutils_enable_sphinx` line for now